device/transport/amqp/src/amqp.ts

```diff
@@ -278,8 +278,11 @@ export class Amqp extends EventEmitter implements DeviceTransport {
                   this._d2cEndpoint = endpoint.deviceEventPath(credentials.deviceId);
                   this._messageEventName = 'message';
                 }
-
-                getUserAgentString((userAgentString) => {
+                let customInfo = '';
+                if (this.hasOwnProperty('_options') && this._options.hasOwnProperty('productInfo')) {
```

Comment:
>this.hasOwnProperty('_options') && this._options.hasOwnProperty('productInfo') [](start = 20, length = 78)

const customInfo = (this._options && this._options.productInfo) ? this._options._productInfo : undefined;